task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
+from pyrsistent import pmap
+from pyrsistent import PVector
+from pyrsistent import pvector
+from pyrsistent import v
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+VOLUME_KEYS = set(['mode', 'container_path', 'host_path'])
+
+def valid_volumes(volumes):
+    for vol in volumes:
+        if set(vol.keys()) != VOLUME_KEYS:
+            return (
+                False,
+                'Invalid volume format, must only contain following keys: '
+                '{}, was: {}'.format(VOLUME_KEYS, vol.keys())
+            )
+    return (True, None)
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    uuid = field(type=(str, uuid.UUID), initial=uuid.uuid4)
+    node_name = field(type=str)
+    node_selector = field(type=(str,str))
```

Comment:
I think we can probably want to store this in a closer form to how k8s does, so this should be a [PMap](https://pypi.org/project/pyrsistent/#pmap) (unless you see an opportunity to do some validation here, in which case we'd want a PRecord)

Reply:
Changed to PMap.